app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -436,29 +436,30 @@ function() {
      * Get address data for sending as hints.
      *
      * @param $session      Customer session
+     * @param string $checkoutType  'multi-page' | 'one-page' | 'admin'
+     *
      * @return array        hints data
      */
-    private function getAddressHints($session, $quote)
+    private function getAddressHints($session, $quote, $checkoutType)
     {
 
         $hints = array();
 
-        ///////////////////////////////////////////////////////////////
+        /////////////////////////////////////////////////////////////////////////
         // Check if the quote shipping address is set,
         // otherwise use customer shipping address for logged in users.
-        ///////////////////////////////////////////////////////////////
+        /////////////////////////////////////////////////////////////////////////
         $address = $quote->getShippingAddress();
-
-        if ($session && $session->isLoggedIn()) {
-            /** @var Mage_Customer_Model_Customer $customer */
-            $customer = Mage::getModel('customer/customer')->load($session->getId());
-            $address = $customer->getPrimaryShippingAddress();
-            $hints['email'] = $customer->getEmail();
+        if (!$address->getStreet1()) {
```

Comment:
what is this check for?

Reply:
https://github.com/BoltApp/bolt-magento1/blob/af9f8d6f24c62efc27a415d13bec0c644afd584d/app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php#L448-L454

The checks do exactly what the comments above say that they do.
The first checks if there is a valid shipping address set in the quote instead of an empty address with null values.  This is accomplished by checking if the mandatory field of *street1* is set.

If this is not set, then the next line check if the customer is logged in.  If so, then we get the address data from the primary shipping address saved by the customer.

https://github.com/BoltApp/bolt-magento1/blob/af9f8d6f24c62efc27a415d13bec0c644afd584d/app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php#L463-L473

line 463, as I read it, will always evaluate to true.  Additionally, the `@` error message suppression operators in lines 464-473 are also unnecessary, as they are called on a `Varien_Object` that has a magic __get method that handles this and will return null without message.  I would normally remove both of these to get the pull request merged more quickly, but, I think now that it is better to make the changes and answer all questions which arise in order to continually improve the codebase.

Reply:
If we know address is always non-null, and you make change here at line 453 to account for that, let's cleanup line 463 as well.